job-shop-scheduling/job_shop_scheduler.py

```diff
@@ -319,18 +313,46 @@ def get_bqm(self, stitch_kwargs=None):
         for i in self.last_task_indices:
             task = self.tasks[i]
 
-            for t in range(self.max_time):
-                end_time = t + task.duration
+            for t in range(self.max_time + 1):
+                end_time = t + task.duration - 1    # -1 to get last unit of time the task occupies
 
-                # Check task's end time; do not add in absurd times
-                if end_time > self.max_time:
+                # Check task's end time
+                # Note: first condition is to prevent adding in absurd times. Second condition is
+                #   to prevent penalizing job end-times shorter than the shortest possible schedule
+                #   end-time (i.e. the time it takes to run the longest job).
+                if end_time > self.max_time or end_time <= self.max_job_time:
                     continue
 
                 # Add bias to variable
+                # Note: the bias shown here is a scaled version of the proof shown above. Rather
+                #   than simply doing base**end_time, I have scaled the all biases with
+                #   2 / base**self.max_time. This way, the largest possible bias
+                #   (when end_time==(self.max_time-1)) is 2.
                 bias = 2 * base**(end_time - self.max_time)
                 label = get_label(task, t)
                 if label in pruned_variables:
                     bqm.add_variable(label, bias)
 
+    def get_bqm(self, stitch_kwargs=None):
+        """Returns a BQM to the Job Shop Scheduling problem.
+
+        Args:
+            stitch_kwargs: A dict. Kwargs to be passed to dwavebinarycsp.stitch.
+        """
+        if stitch_kwargs == None:
```

Comment:
`is` operator is preferred for comparison to `None` ([PEP-8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)).

Imagine `stitch_kwargs` implements a custom `__eq__`...